proto/osmosis/gamm/v1beta1/query.proto

```diff
@@ -172,3 +178,23 @@ message QueryTotalLiquidityResponse {
     (gogoproto.nullable) = false
   ];
 }
+
+//=============================== PoolTwap
+message QueryPoolTwapRequest {
+  uint64 poolId = 1 [ (gogoproto.moretags) = "yaml:\"pool_id\"" ];
+  string tokenInDenom = 2 [ (gogoproto.moretags) = "yaml:\"token_in_denom\"" ];
+  string tokenOutDenom = 3
+      [ (gogoproto.moretags) = "yaml:\"token_out_denom\"" ];
```

Comment:
Should these be optional, OR should there be a query to get all twap responses?

Reply:
hmmm...What do you refer to by all twap responses?

Reply:
Like I want to get all the TWAP prices for a pool

Reply:
TODO: Write in a second PR later